File: src/main/java/guru/springframework/spring6webapp/service/AuthorService.java
package guru.springframework.spring6webapp.service;

import guru.springframework.spring6webapp.entity.Author;
import guru.springframework.spring6webapp.repository.AuthorRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class AuthorService {
    private AuthorRepository authorRepository;

    public Iterable<Author> findAll() {
        return authorRepository.findAll();
    }

    public Author findById(Long id) {
        return authorRepository.findById(id).orElseGet(Author::new);
    }

}

File: src/main/kotlin/guru/springframework/spring6webapp/service/AuthorService.kt
package guru.springframework.spring6webapp.service

import guru.springframework.spring6webapp.entity.Author
import guru.springframework.spring6webapp.repository.AuthorRepository
import org.springframework.stereotype.Service


@Service
class AuthorService(private val authorRepository: AuthorRepository) {

    fun findAll(): Iterable<Author> = authorRepository.findAll()

    fun findById(id: Long): Author = authorRepository.findById(id).orElseGet { Author() }
}